packages/perseus-editor/src/widgets/interactive-graph-editor/interactive-graph-editor.tsx

```diff
@@ -294,17 +291,16 @@ class InteractiveGraphEditor extends React.Component<Props> {
                 showTooltips: this.props.showTooltips,
                 lockedFigures: this.props.lockedFigures,
                 trackInteraction: function () {},
-                onChange: (newProps: InteractiveGraphProps) => {
+                onChange: ({graph: newGraph}: InteractiveGraphProps) => {
                     let correct = this.props.correct;
-                    // @ts-expect-error - TS2532 - Object is possibly 'undefined'.
-                    if (correct.type === newProps.graph.type) {
-                        correct = {
-                            ...correct,
-                            ...newProps.graph,
-                        };
+                    // TODO(benchristel): can we improve the type of onChange
+                    // so this invariant isn't necessary?
+                    invariant(newGraph != null);
```

Comment:
Is it better to use `invariant` than to do a direct check for existence? Not asking for a change, I'm just curious because I haven't really seen this `invariant` pattern used in other places.